code/controllers/subsystem/economy.dm

```diff
@@ -48,46 +75,78 @@ SUBSYSTEM_DEF(economy)
+		if(account_bitflags[B] == target_bitflag)
+			return B
+
+/// How to use: `bankaccount.is_nonstation_account()`, `SSeconomy.is_nonstation_account(bankaccount type)` or `SSecononmy.is_nonstation_account(ACCOUNT_CAR)`. This returns TRUE if an account is non-station account - i.e.) VIP budget, Command budget
+/datum/controller/subsystem/economy/proc/is_nonstation_account(datum/bank_account/department/D) // takes a bank account type or dep_ID define
+	if(!D) // null check first
+		return FALSE
+	if(!istype(D, /datum/bank_account/department)) // if parameter was given as a dept id, replace it into better type
+		D = SSeconomy.get_dep_account(D) // tricky
+	if(!istype(D, /datum/bank_account/department)) // if it failed to replacing, return false.
+		return FALSE
+	for(var/each in SSeconomy.nonstation_accounts)
+		if(D.account_holder == SSeconomy.nonstation_accounts[each])
+			return TRUE
+	return FALSE
+
 /datum/controller/subsystem/economy/proc/distribute_funds(amount)
-	var/datum/bank_account/eng = get_dep_account(ACCOUNT_ENG)
-	var/datum/bank_account/sec = get_dep_account(ACCOUNT_SEC)
-	var/datum/bank_account/med = get_dep_account(ACCOUNT_MED)
-	var/datum/bank_account/srv = get_dep_account(ACCOUNT_SRV)
-	var/datum/bank_account/sci = get_dep_account(ACCOUNT_SCI)
-	var/datum/bank_account/civ = get_dep_account(ACCOUNT_CIV)
-	var/datum/bank_account/car = get_dep_account(ACCOUNT_CAR)
+	var/datum/bank_account/eng = get_dep_account(ACCOUNT_ENG_ID)
+	var/datum/bank_account/sec = get_dep_account(ACCOUNT_SEC_ID)
+	var/datum/bank_account/med = get_dep_account(ACCOUNT_MED_ID)
+	var/datum/bank_account/srv = get_dep_account(ACCOUNT_SRV_ID)
+	var/datum/bank_account/sci = get_dep_account(ACCOUNT_SCI_ID)
+	var/datum/bank_account/civ = get_dep_account(ACCOUNT_CIV_ID)
+	var/datum/bank_account/car = get_dep_account(ACCOUNT_CAR_ID)
 
 	var/departments = 0
 
 	if(eng)
-		departments += 2
+		departments += BUDGET_RATIO_TYPE_B
 	if(sec)
-		departments += 2
+		departments += BUDGET_RATIO_TYPE_B
 	if(med)
-		departments += 2
+		departments += BUDGET_RATIO_TYPE_B
 	if(srv)
-		departments += 1
+		departments += BUDGET_RATIO_TYPE_A
 	if(sci)
-		departments += 2
+		departments += BUDGET_RATIO_TYPE_B
 	if(civ)
-		departments += 1
+		departments += BUDGET_RATIO_TYPE_A
 	if(car)
-		departments += 2
+		departments += BUDGET_RATIO_TYPE_B
 
 	var/parts = round(amount / departments)
 
-	var/engineering_cash = parts * 2
-	var/security_cash = parts * 2
-	var/medical_cash = parts * 2
-	var/service_cash = parts
-	var/science_cash = parts * 2
-	var/civilian_cash = parts
-	var/cargo_cash = parts * 2
+	var/engineering_cash = parts * BUDGET_RATIO_TYPE_B
+	var/security_cash = parts * BUDGET_RATIO_TYPE_B
+	var/medical_cash = parts * BUDGET_RATIO_TYPE_B
+	var/service_cash = parts * BUDGET_RATIO_TYPE_A
+	var/science_cash = parts * BUDGET_RATIO_TYPE_B
+	var/civilian_cash = parts * BUDGET_RATIO_TYPE_A
+	var/cargo_cash = parts * BUDGET_RATIO_TYPE_B
 
 	eng?.adjust_money(engineering_cash)
```

Comment:
this is terrible and needs to be made into a for loop over all the department budgets instead of a bunch of hardcoded garbage